doc/api/net.md

```diff
@@ -246,168 +352,217 @@ server.on('error', (e) => {
-let the program exit if it's the only server left (the default behavior). If
-the server is `ref`d calling `ref` again will have no effect.
+Calling the `server.ref()` method will cause the Node.js event loop to continue
+so long as the `server` is not closed. Multiple calls to `server.ref()` will
+have no effect.
 
-Returns `server`.
+Returns a reference to `server`.
 
 ### server.unref()
 <!-- YAML
 added: v0.9.1
 -->
 
-Calling `unref` on a server will allow the program to exit if this is the only
-active server in the event system. If the server is already `unref`d calling
-`unref` again will have no effect.
+Calling the `server.unref()` method ensures that the Node.js event loop will
+not be required to continue if this `server` is still open. Multiple calls to
+`server.unref()` will have no effect.
 
-Returns `server`.
+Returns a reference to `server`.
 
 ## Class: net.Socket
 <!-- YAML
 added: v0.3.4
 -->
 
-This object is an abstraction of a TCP or local socket.  `net.Socket`
-instances implement a duplex Stream interface.  They can be created by the
-user and used as a client (with [`connect()`][]) or they can be created by Node.js
-and passed to the user through the `'connection'` event of a server.
+The `net.Socket` object is an abstraction of a TCP or local socket. All
+`net.Socket` instances implement a [Duplex][] Stream interface.
+
+Instances may be created by users (e.g. using [`net.connect()`][]) or by
+Node.js and passed to the user through the `'connection'` event.
+
+All `net.Socket` objects are [`EventEmitter`][] instances.
 
 ### new net.Socket([options])
 <!-- YAML
 added: v0.3.4
 -->
 
-Construct a new socket object.
+* `options` {Object}
+  * `fd` {number} Used to specify a numeric file descriptor of an existing
+    socket. Defaults to `null`.
+  * `allowHalfOpen` {boolean} If `true`, allows the socket to be in a "half
+    open" state. Defaults to `false`.
+  * `readable` {boolean} If `fd` is specified, setting `readable` to `true`
+    allows data from the socket to be read. Defaults to `false`.
+  * `writable` {boolean} If `fd` is specified, setting `writabl` to `true`
+    allows data to be written to the socket. Defaults to `false`.
 
-`options` is an object with the following defaults:
-
-```js
-{
-  fd: null,
-  allowHalfOpen: false,
-  readable: false,
-  writable: false
-}
-```
-
-`fd` allows you to specify the existing file descriptor of socket.
-Set `readable` and/or `writable` to `true` to allow reads and/or writes on this
-socket (NOTE: Works only when `fd` is passed).
-About `allowHalfOpen`, refer to `createServer()` and `'end'` event.
-
-`net.Socket` instances are [`EventEmitter`][] with the following events:
+Construct a new `net.Socket` instance.
 
 ### Event: 'close'
 <!-- YAML
 added: v0.1.90
 -->
 
-* `had_error` {Boolean} `true` if the socket had a transmission error.
+The `'close'` event is emitted once the socket is closed and can no longer be
+used to send or receive data.
```

Comment:
if half open is allowed, doesn't it just mean no more data will be read, but its still writable?